src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseRangeOperatorDiagnosticAnalyzer.cs

```diff
@@ -124,12 +125,22 @@ protected override void InitializeWorker(AnalysisContext context)
             // Second arg needs to be a subtraction for: `end - e2`.  Once we've seen that we have
             // that, try to see if we're calling into some sort of Slice method with a matching
             // indexer or overload
+            var valid = infoCache.TryGetMemberInfo(targetMethod, out var memberInfo);
             if (!IsSubtraction(invocation.Arguments[1].Value, out var subtraction) ||
-                !infoCache.TryGetMemberInfo(targetMethod, out var memberInfo))
+                !valid)
             {
                 return null;
             }
 
+            // Need to make sure that if we are assigning values to the slice that both the slice-like method and the
+            // indexer are references
+            if (invocation.Syntax.IsLeftSideOfAnyAssignExpression() &&
+                !valid)
```

Comment:
don't need multiple checks of `!valid`.  can be hoisted once after the TryGetMemberInfo right?